Let the Tooling API control system properties contributed from the client-side (#21223)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

```

> ### IDE integration improvements

Reviewer: ```suggestion
```
Suggestion: No need for a new section, this fits just fine under the "IDE" section.

> #### Tooling API can control system properties passed to the build

Reviewer: ```suggestion
#### Added support for passing system properties to the build with the Tooling API
```
Suggestion: rephrase for consistency with other section titles in the release notes.

Author: Thank you for the help, @nathan-contino!

> Before 7.6, builds started from the Tooling API with the system properties from the host JVM, leaking configuration from the IDE to the build.
> Gradle 7.6 introduced `LongRunningOperation.withSystemProperties(Map)` to provide an isolated set of system properties to the build. 
> For more details, check the [`javadoc`](javadoc/org/gradle/tooling/LongRunningOperation.html#withSystemProperties-java.util.Map-).

Reviewer: ```suggestion
Before 7.6, the Tooling API started builds with the system properties from the host JVM. This leaked configuration from the IDE to the build.
Starting in Gradle 7.6, `LongRunningOperation.withSystemProperties(Map)` provides
an isolated set of build system properties. 
For more information, see [`LongRunningOperation`](javadoc/org/gradle/tooling/LongRunningOperation.html#withSystemProperties-java.util.Map-).
```
Suggestion: minor rephrases for consistency with other new feature sections.
